src/test/java/seedu/address/logic/LogicManagerTest.java

```diff
@@ -94,11 +98,39 @@ public void execute_storageThrowsIoException_throwsCommandException() {
         assertCommandFailure(addCommand, CommandException.class, expectedMessage, expectedModel);
     }
 
+    @Test
+    public void execute_projectsFolderStorageThrowsIoException_throwsCommandException() {
+        // Setup LogicManager with JsonProjectsFolderIoExceptionThrowingStub
+        JsonAddressBookStorage addressBookStorage =
+                new JsonAddressBookStorage(temporaryFolder.resolve("ioExceptionAddressBook.json"));
+        JsonProjectsFolderStorage projectsFolderStorage =
+                new JsonProjectsFolderIoExceptionThrowingStub(
+                temporaryFolder.resolve("ioExceptionProjectsFolder.json"));
+        JsonUserPrefsStorage userPrefsStorage =
+                new JsonUserPrefsStorage(temporaryFolder.resolve("ioExceptionUserPrefs.json"));
+        StorageManager storage = new StorageManager(addressBookStorage, projectsFolderStorage, userPrefsStorage);
+        logic = new LogicManager(model, storage);
+
+        // Execute add command
+        String projectName = "New Project";
+        String addCommand = AddProjectCommand.COMMAND_WORD + " " + CliSyntax.PREFIX_NAME + projectName;
+        Project expectedProject = new ProjectBuilder().withName(projectName).build();
+        ModelManager expectedModel = new ModelManager();
+        expectedModel.addProject(expectedProject);
+        String expectedMessage = LogicManager.FILE_OPS_ERROR_MESSAGE + DUMMY_IO_EXCEPTION;
+        assertCommandFailure(addCommand, CommandException.class, expectedMessage, expectedModel);
+    }
+
     @Test
     public void getFilteredPersonList_modifyList_throwsUnsupportedOperationException() {
         assertThrows(UnsupportedOperationException.class, () -> logic.getFilteredPersonList().remove(0));
```

Comment:
Looks repeated as the one below.

Reply:
Not repeated one for persons one for projects